src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,294 @@
+.. ## Copyright (c) 2017-2022, Lawrence Livermore National Security,con LLC and
```

Comment:
Since this is a porting of the document, not the writing, this comment may not apply.  But as a newbie reading this doc, I have one comment.  This page does a good job of introducing me to what I have to do to port to GPU, but to write code, I'd have to get more details from the header files.  Some references to where the headers and how they are organized into directories would be helpful.